Packs/CommonPlaybooks/Playbooks/playbook-Search_And_Delete_Emails_-_Generic_-_v2_6_0.yml

```diff
@@ -0,0 +1,1067 @@
+      max: 100
+    view: |-
+      {
+        "position": {
+          "x": -1620,
+          "y": 1230
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: true
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "15":
+    id: "15"
+    taskid: bfa45743-9c08-4aba-8df6-cbff71d46bed
+    type: title
+    task:
+      id: bfa45743-9c08-4aba-8df6-cbff71d46bed
+      version: -1
+      name: Agari Phishing Defense
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    nexttasks:
+      '#none#':
+      - "16"
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 1530,
+          "y": 440
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "16":
+    id: "16"
+    taskid: b622809c-289a-413c-8019-3d5bafda95cc
+    type: regular
+    task:
+      id: b622809c-289a-413c-8019-3d5bafda95cc
+      version: -1
+      name: Search And Delete Emails - Agari Phishing Defense
+      description: Remediate suspected message.
+      script: Agari Phishing Defense|||apd-remediate-message
+      type: regular
+      iscommand: true
+      brand: Agari Phishing Defense
+    nexttasks:
+      '#none#':
+      - "2"
+    scriptarguments:
+      id:
+        complex:
+          root: incident
+          accessor: emailinternalmessageid
+      operation:
+        simple: delete
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 1530,
+          "y": 1230
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: true
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+system: true
+view: |-
+  {
+    "linkLabelsPosition": {
+      "4_13_Gmail": 0.81,
+      "4_15_Agari Phishing Defense": 0.86,
+      "4_2_#default#": 0.9
+    },
+    "paper": {
+      "dimensions": {
+        "height": 1435,
+        "width": 3530,
+        "x": -1620,
+        "y": 50
+      }
+    }
+  }
+inputs:
+- key: From
+  value:
+    complex:
+      root: incident
+      accessor: emailfrom
+  required: false
+  description: The value of the malicious email's "From" attribute.
+  playbookInputQuery:
+- key: Subject
+  value:
+    complex:
+      root: incident
+      accessor: emailsubject
+  required: false
+  description: The value of the malicious email's "Subject" attribute.
+  playbookInputQuery:
+- key: AttachmentName
+  value:
+    complex:
+      root: incident
+      accessor: attachmentname
+  required: false
+  description: The value of the malicious email's "AttachmentName" attribute.
+  playbookInputQuery:
+- key: SearchAndDeleteIntegration
+  value: {}
+  required: true
+  description: The integration in which to run the search and delete action. Can be
+    O365, or EWS.
```

Comment:
We support additional integrarions...

Reply:
Also, why is it a mandatory input?

Reply:
Legacy + the need for special permissions to do so. running on all integrations will raise errors.